### PR TITLE
WIP Carter/vector

### PR DESCRIPTION
**Pull request by cartazio**

### CODE REVIEW

**Review by VictorOdede — COMMENTED** [review]
we need to pass `self` into the class methods

**Comment by cartazio** [comment]
derp :)

File: src/llm_vm/vector_store.py
[the remark this answers is not in the document]
Comment (cartazio) [reply src/llm_vm/vector_store.py:1]: It’s a bad idea I guess.  I just felt inspired :)

**Comment by cartazio** [comment]
this PR doesn't have the "db/storage handle" objects, but those are by necessity user provided if we want an easy to use abstraction

**Comment by cartazio** [comment]
also we need to come up with a sane interface for how this wrapps up a base context

**Comment by cartazio** [comment]
@VictorOdede  should be ready to merge, but have a look see before I do that :)

File: src/llm_vm/vector_store.py
Comment (VictorOdede) [inline src/llm_vm/vector_store.py:1]: Looks good to me. Just curious why `default_merge` function is defined outside the `WrappedRAG` class?

**Pull request closed without merging**